tests/qemu-tests/download_images.sh

```diff
@@ -8,21 +8,23 @@ URL="https://github.com/gsingh93/linux-exploit-dev-env/releases/latest/download"
 
 mkdir -p "${OUT_DIR}"
 
-for arch in x86_64 arm64; do
-    file="rootfs-${arch}.img"
-    wget "${URL}/${file}" -O "${OUT_DIR}/${file}"
-
-    file="vmlinux-linux-${arch}"
-    wget "${URL}/${file}" -O "${OUT_DIR}/${file}"
-
-    file="vmlinux-ack-${arch}"
-    wget "${URL}/${file}" -O "${OUT_DIR}/${file}"
-done
-
-for kernel_type in ack linux; do
-    file="bzImage-${kernel_type}-x86_64"
-    wget "${URL}/${file}" -O "${OUT_DIR}/${file}"
-
-    file="Image-${kernel_type}-arm64"
-    wget "${URL}/${file}" -O "${OUT_DIR}/${file}"
-done
+download() {
+    file="$1"
+    hash_old=$(grep "${file}" "${OUT_DIR}/hashsums.txt.old" || true)
+    hash_new=$(grep "${file}" "${OUT_DIR}/hashsums.txt")
+    # only download file if it doesn't exist or its hashsum has changed
+    if [ ! -f "${OUT_DIR}/${file}" ] || [ "${hash_new}" != "${hash_old}"]; then
```

Comment:
Don't you need a space at the end here: `[ "${hash_new}" != "${hash_old}"]` ? I'm surprised this worked.

Reply:
you're right. interesting.